src/main/java/org/apache/mesos/hdfs/Scheduler.java

```diff
@@ -76,15 +77,185 @@ public void registered(SchedulerDriver driver, FrameworkID frameworkId, MasterIn
       throw new RuntimeException(e);
     }
     log.info("Registered framework frameworkId=" + frameworkId.getValue());
+    liveState.updateReconciliationTimestamp();
+    driver.reconcileTasks(Collections.<Protos.TaskStatus>emptyList());
   }
-
   @Override
   public void reregistered(SchedulerDriver driver, MasterInfo masterInfo) {
-    log.info("Reregistered framework.");
+    log.info("Reregistered framework: starting task reconcile");
+    liveState.updateReconciliationTimestamp();
+    driver.reconcileTasks(Collections.<Protos.TaskStatus> emptyList());
+  }
+
+  @Override
+  public void statusUpdate(SchedulerDriver driver, TaskStatus status) {
+    log.info(String.format(
+        "Received status update for taskId=%s state=%s message='%s' stagingTasks.size=%d",
+        status.getTaskId().getValue(),
+        status.getState().toString(),
+        status.getMessage(),
+        liveState.getStagingTasksSize()));
+
+    if (!isStagingState(status)) {
+      liveState.removeStagingTask(status.getTaskId());
+    }
+
+    if (isTerminalState(status)) {
+      liveState.removeRunningTask(status.getTaskId());
+      persistentState.removeTaskId(status.getTaskId().getValue());
+      if (liveState.reconciliationComplete()) {
+        correctCurrentPhase();
+      }
+
+    } else if (isRunningState(status)) {
+      liveState.updateTaskForStatus(status);
+
+      log.info(String.format("Current Acquisition Phase: %s", liveState
+          .getCurrentAcquisitionPhase().toString()));
+
+      switch (liveState.getCurrentAcquisitionPhase()) {
+        case RECONCILING_TASKS :
+          if (liveState.reconciliationComplete()) {
+            reconcilePersistentState();
+            correctCurrentPhase();
+          }
+          break;
+        case JOURNAL_NODES :
+          if (liveState.getJournalNodeSize() == conf.getJournalNodeCount()) {
+            correctCurrentPhase();
+          }
+          break;
+        case START_NAME_NODES :
+          if (liveState.getNameNodeSize() == (HDFSConstants.TOTAL_NAME_NODES)) {
+            // TODO move the reload to correctCurrentPhase and make it idempotent
+            reloadConfigsOnAllRunningTasks(driver);
+            correctCurrentPhase();
+          }
+          break;
+        case FORMAT_NAME_NODES :
+          if (!liveState.isNameNode1Initialized()
+              && !status.getMessage().equals(HDFSConstants.NAME_NODE_INIT_MESSAGE)) {
```

Comment:
Why do you need to check for INIT_MESSAGE? Wouldn't `liveState.updateTaskForStatus(status);` on line 111 have modified `nameNode1TaskMap` such that `isNameNode1Initialized()` returns true?